c7n/resources/directory.py

```diff
@@ -172,12 +172,12 @@ class DirectoryTrustFilter(ValueFilter):
     def process(self, resources, event=None):
         client = local_session(self.manager.session_factory).client('ds')
         trusts = client.describe_trusts()['Trusts']
+        matched = []
         for r in resources:
             r[self.annotation_key] = [
                 t for t in trusts if t['DirectoryId'] == r['DirectoryId']]
-        matched = []
-        for r in resources:
-            if any((self.match(trust) for trust in r[self.annotation_key])):
+            resource_trusts = r[self.annotation_key] if len(r[self.annotation_key]) >= 1 else [{}]
```

Comment:
the use of the empty dict (aka list with 1 item) here feels a little policy specific, would appreciate a comment inline to distinguish purpose.